internal/dcache/rpc/response_error.go

```diff
@@ -117,9 +141,10 @@ func IsTimedOut(err error) bool {
 		return false
 	}
 
-	// TODO: This is untested, see whether this works or the ETIMEDOUT check works!
-	//te := thrift.NewTTransportExceptionFromError(err)
-	//return te.TypeId() == thrift.TIMED_OUT
+	te := thrift.NewTTransportExceptionFromError(err)
+	log.Debug("IsTimedOut: err: %v, err: %T, te.TypeId(): %d, Is syscall.ETIMEDOUT: %v",
+		err, err, te.TypeId(), errors.Is(err, syscall.ETIMEDOUT))
 
-	return errors.Is(err, syscall.ETIMEDOUT)
+	// TODO: See which one of these works.
+	return te.TypeId() == thrift.TIMED_OUT || errors.Is(err, syscall.ETIMEDOUT)
 }
```

Comment:
Can we also please add a comment above this method, `IsTimedOut(err error)`, describing when this will be called, for example when the node is down, we will get timeout error.